pallets/parachain-staking/src/tests/bond_extra_tests.rs

```diff
@@ -18,6 +18,232 @@ fn to_acc_id(id: u64) -> AccountId {
+
+                // Pick an amount that is not perfectly divisible by the number of collators
+                let dust = 1u128;
+                let amount_to_topup = (min_user_stake * 2u128) + dust;
+                let nonce = ParachainStaking::proxy_nonce(staker.account_id);
+                let bond_extra_call = create_call_for_bond_extra(&staker, nonce, amount_to_topup);
+                assert_ok!(AvnProxy::proxy(Origin::signed(staker.relayer), bond_extra_call, None));
+
+                // The staker state has also been updated
+                let staker_state = ParachainStaking::nominator_state(staker.account_id).unwrap();
+                assert_eq!(staker_state.total(), initial_stake * 2 + amount_to_topup);
+
+                // Each collator has been topped up by the expected amount
+                for (index, collator) in collators.into_iter().enumerate() {
+                    // We should have one event per collator. One of the collators gets any
+                    // remaining dust.
+                    let mut topup = min_user_stake;
+                    if index == 1 {
+                        topup = min_user_stake + dust;
+                    }
+
+                    assert_event_emitted!(Event::NominationIncreased {
+                        nominator: staker.account_id,
+                        candidate: collator,
+                        amount: topup,
+                        in_top: true
+                    });
+
+                    // Staker state reflects the new nomination for each collator
+                    assert_eq!(staker_state.nominations.0[index].owner, collator);
+                    assert_eq!(staker_state.nominations.0[index].amount, initial_stake + topup);
+
+                    // Collator state has been updated
+                    let collator_state = ParachainStaking::candidate_info(collator).unwrap();
+                    assert_eq!(collator_state.total_counted, initial_stake + initial_stake + topup);
+
+                    // Collator nominations have also been updated
+                    let top_nominations = ParachainStaking::top_nominations(collator).unwrap();
+                    assert_eq!(top_nominations.total, initial_stake + topup);
+                }
+
+                // The staker free balance has been reduced
+                assert_eq!(
+                    ParachainStaking::get_nominator_stakable_free_balance(&staker.account_id),
+                    10000 - (initial_stake * 2 + amount_to_topup)
+                );
+
+                // The total amount staked on chain should increase
+                assert_eq!(
+                    initial_total_stake_on_chain + amount_to_topup,
+                    ParachainStaking::total()
+                );
+
+                // Nonce has increased
+                assert_eq!(ParachainStaking::proxy_nonce(staker.account_id), nonce + 1);
+            })
+    }
+
+    mod fails_when {
```

Comment:
:100: 

Several tests include control assertions that isolate a single failure reason.
That is great! :dart: